src/javascript/components/SitemapPanelHeader/SitemapPanelHeader.jsx

```diff
@@ -4,21 +4,70 @@ import {Book, Button, Delete, Header, Save, Upload} from '@jahia/moonstone';
 import styles from './SitemapPanelHeader.scss';
 import {DialogComponent} from '../Dialog/Dialog';
 import {useTranslation} from 'react-i18next';
+import {useMutation} from '@apollo/react-hooks';
+
+import * as gqlMutations from './gqlMutations';
 
 export const SitemapPanelHeaderComponent = ({
     formik,
     isSitemapMixinEnabled,
-    siteKey
+    siteKey,
+    snackBarInfo,
+    openSnackBar
 }) => {
     const {t} = useTranslation('sitemap');
     const [dialogIsOpen, setDialogIsOpen] = useState(false);
     const [dialogInfo, setDialogInfo] = useState(null);
 
-    const handleDialogOpen = (title, text, submitText) => {
+    const [submitToGoogleMutation] = useMutation(gqlMutations.sendSitemapToSearchEngine, {
+        variables: {
+            sitemapURL: formik.values.sitemapIndexURL
+        },
+        // eslint-disable-next-line no-unused-vars
+        onCompleted: data => {
+            snackBarInfo(t('labels.snackbar.successSubmitToGoogle'));
+            openSnackBar(true);
+            handleDialogClose();
+        },
+        // eslint-disable-next-line no-unused-vars
+        onError: error => {
+            handleDialogClose();
+        }
+    });
+
+    const [deleteSitemapCacheMutation] = useMutation(gqlMutations.deleteSitemapCache, {
+        variables: {
+            expirationTimeDifference: formik.values.sitemapCacheDuration
+        },
+        // eslint-disable-next-line no-unused-vars
+        onCompleted: data => {
+            snackBarInfo(t('labels.snackbar.successFlushCache'));
+            openSnackBar(true);
+            handleDialogClose();
+        },
+        // eslint-disable-next-line no-unused-vars
+        onError: error => {
+            handleDialogClose();
+        }
+    });
+
+    const handleDialogOpen = (id, title, text, submitText) => {
+        let submitFunc = null;
+        if (id === 'flushCache') {
+            submitFunc = () => {
+                deleteSitemapCacheMutation();
+            };
+        } else if (id === 'submitToGoogle') {
+            submitFunc = () => {
```

Comment:
fyi you might be able to just do `submitFunc = submitToGoogleMutation` depending on context

Reply:
You are right (I will clean that up)